scipy/stats/tests/test_hypotests.py

```diff
@@ -899,28 +899,30 @@ def test_somersd_alternative(self):
         with pytest.raises(ValueError, match="alternative must be 'less'..."):
             stats.somersd(x1, x2, alternative="ekki-ekki")
 
-    def test_somersd_perfect_correlation(self):
+    @pytest.mark.parametrize("positive_correlation", (False, True))
+    def test_somersd_perfect_correlation(self, positive_correlation):
         # Before the addition of `alternative`, perfect correlation was
         # treated as a special case. Now it is treated like any other case, but
         # make sure there are no divide by zero warnings or associated errors
 
         x1 = np.arange(10)
-        x2 = x1
+        x2 = x1 if positive_correlation else np.flip(x1)
+        expected_statistic = 1 if positive_correlation else -1
 
         # perfect correlation -> small "two-sided" p-value (0)
         res = stats.somersd(x1, x2, alternative="two-sided")
-        assert res.statistic == 1
+        assert res.statistic == expected_statistic
         assert res.pvalue == 0
 
         # rank correlation > 0 -> large "less" p-value (1)
```

Comment:
this comment does not hold with the new change. How about changing this to:

```
# perfect correlation -> large "less" p-value (1)
# perfect negative correlation -> small "less" p-value (0)
```

Reply:
Ah, I get it. Fine, the comment is clear enough to figure out the `rank correlation < 0` part.